edb/pgsql/metaschema.py

```diff
@@ -2172,12 +2167,12 @@ class StringSliceImplFunction(dbops.Function):
                 pg_end - pg_start
             )
         FROM (SELECT
-            GREATEST(0, edgedb._int_bound(edgedb._normalize_array_index(
-                start, edgedb._length(val)
-            ))) as pg_start,
-            GREATEST(0, edgedb._int_bound(edgedb._normalize_array_index(
-                stop, edgedb._length(val)
-            ))) as pg_end
+            GREATEST(0, edgedb._normalize_array_index(
+                edgedb._int_bound(start), edgedb._length(val)
+            ))::int as pg_start,
+            GREATEST(0, edgedb._normalize_array_index(
+                edgedb._int_bound(stop), edgedb._length(val)
+            ))::int as pg_end
```

Comment:
`GREATEST` is not needed anymore. 

Now the output of `_normalize_array_index` is on interval `[0, 2147483647]`. Before the output of `_int_bound` was `[-2147483648, 2147483647]`, because `_normalize` could produce very negative numbers. See my comment here: https://github.com/edgedb/edgedb/pull/4448#discussion_r983950307